### PR TITLE
flatbuffers: unbreak with gcc10

### PR DESCRIPTION
**Pull request by rmcgibbo**
###### Motivation for this change
Hydra failures -- https://hydra.nixos.org/job/nixpkgs/trunk/flatbuffers.x86_64-linux. I think because of an upgrade of gcc but I didn't bisect all the way.

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by teh — APPROVED** [review]
thanks!

**Comment by austinbutler** [comment]
Fixes the build for me.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108415` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>flatbuffers</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108415` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
24 packages marked as broken and skipped:
</summary>
<ul>
<li>kodiPlugins.inputstream-adaptive</li>
<li>kodiPlugins.joystick</li>
<li>kodiPlugins.pvr-hdhomerun</li>
<li>kodiPlugins.pvr-hts</li>
<li>kodiPlugins.pvr-iptvsimple</li>
<li>kodiPlugins.steam-controller</li>
<li>kodiPlugins.vfs-libarchive</li>
<li>kodiPlugins.vfs-sftp</li>
<li>python37Packages.arviz</li>
<li>python37Packages.baselines</li>
<li>python37Packages.edward</li>
<li>python37Packages.graph_nets</li>
<li>python37Packages.mask-rcnn</li>
<li>python37Packages.optuna</li>
<li>python37Packages.pymc3</li>
<li>python37Packages.rl-coach</li>
<li>python37Packages.tensorflow</li>
<li>python37Packages.tensorflow-build</li>
<li>python37Packages.tensorflow-build_1</li>
<li>python37Packages.tensorflow-probability</li>
<li>python37Packages.tensorflowWithoutCuda</li>
<li>python37Packages.tensorflow_1</li>
<li>python37Packages.tflearn</li>
<li>python38Packages.mask-rcnn</li>
</ul>
</details>
<details>

  <summary>16 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.datasets: log was empty</li>
<li>python37Packages.google_cloud_bigquery: log was empty</li>
<li>python37Packages.ibis-framework: log was empty</li>
<li>python37Packages.intake: log was empty</li>
<li>python37Packages.pyarrow: log was empty</li>
<li>python37Packages.tensorflow_2: log was empty</li>
<li>(python37Packages.tensorflow-build_2): log was empty</li>
<li>python38Packages.datasets: log was empty</li>
<li>python38Packages.google_cloud_bigquery: log was empty</li>
<li>python38Packages.intake: log was empty</li>
<li>python38Packages.pyarrow: log was empty</li>
<li>python38Packages.tensorflow_2: log was empty</li>
<li>(python38Packages.tensorflow-build_2): log was empty</li>
<li>python39Packages.tensorflow_2: log was empty</li>
<li>(python39Packages.tensorflow-build_2): log was empty</li>
<li>tts: log was empty</li>
  </ul>
</details>
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>arrow-cpp</li>
<li>flatbuffers</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 108415` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>python3Packages.pyarrow</li>
</ul>
</details>
